### PR TITLE
FEAT-#6382: Execute bitwise NOT (~) operations on HDK

### LINKED ISSUE #6382: Execute bitwise NOT (~) operations on HDK
Currently HDK does not support the bitwise NOT operation (see https://github.com/intel-ai/hdk/issues/542) and this operation is forced for arrow execution. This behavior should be changed once the operation is implemented in HDK.

### PR DESCRIPTION
**Pull request by AndreyPavlenko**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6382 
- [] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
@AndreyPavlenko should it work with new hdk release 0.9.0?

**Comment by AndreyPavlenko** [comment]
> @AndreyPavlenko should it work with new hdk release 0.9.0?

Yes it should. I've rebased on master - the HDK tests passed, but there are a lot of failures in other tests, for some reason.

**Comment by anmyachev** [comment]
> > @AndreyPavlenko should it work with new hdk release 0.9.0?
> 
> Yes it should. I've rebased on master - the HDK tests passed, but there are a lot of failures in other tests, for some reason.

It's due the new pandas 2.1.2 release. Fixes for it here: https://github.com/modin-project/modin/pull/6685

**Comment by anmyachev** [comment]
@AndreyPavlenko #6685 is merged. Could yo rebase on master branch?

File: modin/experimental/core/execution/native/implementations/hdk_on_native/expr.py [hunk 1]
```diff
@@ -485,7 +485,7 @@ def invert(self) -> "OpExpr":
         OpExpr
             The resulting bitwise inverse expression.
         """
-        return OpExpr("~", [self], self._dtype)
+        return OpExpr("BIT_NOT", [self], self._dtype)
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/expr.py:488]: Do we test this operation on HDK?

Comment (AndreyPavlenko) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/expr.py:488]: Yes, there is a test - https://github.com/modin-project/modin/blob/master/modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py#L2014 . Currently it's executed on arrow due to https://github.com/modin-project/modin/blob/master/modin/experimental/core/execution/native/implementations/hdk_on_native/expr.py#L1007, but this PR enables the HDK execution.

**Review by anmyachev — APPROVED** [review]

**Review by dchigarev — APPROVED** [review]

**Review by YarShev — APPROVED** [review]

**Pull request merged**